integration-libs/cds/src/profiletag/tracking/tracking.service.ts

```diff
@@ -22,14 +20,22 @@ export class TrackingService {
     return factoryFunction;
   }
   trackEvents(): void {
-    merge(
-      this.profileTagPushEventsService.getPushEvents(),
-      this.profileTagLifecycleService.consentGranted()
-    )
+    this.profileTagPushEventsService
+      .getPushEvents()
       .pipe(
-        skipUntil(this.profileTagLifecycleService.consentGranted()),
-        tap((item: ProfileTagPushEvent) => {
-          this.profileTagEventTracker.notifyProfileTagOfEventOccurence(item);
+        withLatestFrom(
+          this.profileTagLifecycleService.consentChanged().pipe(
+            tap((event) => {
+              // always notify of consent changes
+              this.profileTagEventTracker.notifyProfileTagOfEventOccurence(
+                event
+              );
+            })
+          )
+        ),
+        filter(([_event, consentChanged]) => consentChanged.data.granted), //don't notify other events until consent is granted
+        tap(([event]) => {
+          this.profileTagEventTracker.notifyProfileTagOfEventOccurence(event);
```

Comment:
I'm probably missing something, so I'm wondering why calling the `this.profileTagEventTracker.notifyProfileTagOfEventOccurence` again? It's also _always_ called on line 27?

Reply:
The first time is only for consentChanged events.
The 2nd time is for all other events.

So the 2nd time would not contain the consent events.

This is so that all events that are not consent events do not get pushed until after the consent is true. I.E. make sure that tracking does no occur until consent is granted. 

Let me know if this doesn't make sense and we can have a chat